### PR TITLE
feat: metric explorer compare with other metric - part 1

### PR DESCRIPTION
**Pull request by IrakliJani**
Part 1 of: #12676 

### Description:

CleanShot 2024-12-03 at 23 50 35@2x

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx [hunk 1]
```diff
@@ -73,22 +85,21 @@ export const MetricPeekModal: FC<Props> = ({ opened, onClose }) => {
             case MetricExplorerComparison.DIFFERENT_METRIC:
                 return {
                     type: MetricExplorerComparison.DIFFERENT_METRIC,
-                    metricName: metricQuery.data.name,
+                    // TODO: this is hardcoded for now, should be a dropdown in the UI
+                    metricTable: 'orders',
+                    metricName: 'total_non_completed_order_amount',
```
Comment (IrakliJani) [inline packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx:79]: upcoming work for part 2 (ui)

Comment (joaoviana) [reply packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx:79]: and I'm assuming that the legends + tooltips will be addressed in part 2, right?

File: packages/common/src/utils/metricsExplorer.ts [hunk 2]
```diff
@@ -185,20 +191,31 @@ export const getMetricExplorerDataPointsWithCompare = (
...
         ...Object.keys(offsetGroupByCompareMetricRows),
     ]);
 
+    const compareMetricId =
+        comparison.type === MetricExplorerComparison.PREVIOUS_PERIOD
+            ? metricId
+            : getItemId({
+                  table: comparison.metricTable,
+                  name: comparison.metricName,
+              });
```
Comment (joaoviana) [inline packages/common/src/utils/metricsExplorer.ts:211]: maybe we can be more explicit with the enum for this case? so it's not the _default_ if we add more 

can be part 2

File: packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/features/metricsCatalog/components/MetricPeekModal.tsx:79]: yes

File: packages/common/src/utils/metricsExplorer.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/common/src/utils/metricsExplorer.ts:211]: +

**Pull request merged**